### PR TITLE
Update node types for TS 4.9 beta

### PR DESCRIPTION
**Pull request by sandersn**
Add static method AbortSignal.timeout.
If this works, a later version of TS can re-add AbortSignal.abort and node can add support for that too.

I created a ts4.8/ subdirectory for each of v14, v16, and current. I didn't try to share types like before. With 2*3 entries I think it's reasonably maintainable and the old way was very confusing to change.

typesVersions has a weird entry `">4.9.0-a": { "*": ["*"] }` which works around a bug in semver parsing in Typescript. I need to ship a new version of dtslint to allow this but I wanted to get this PR out for people to look at.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
The semver bug is over at https://github.com/microsoft/TypeScript/issues/50909 FWIW

**Comment by rbuckton** [comment]
The minimum prerelease version is `-0`, so `>4.9.0-0` might be what you want.

File: types/node/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "private": true,
+    "types": "index",
+    "typesVersions": {
+        ">4.9.0-a": { "*": ["*"] },
+        "<=4.8": { "*": ["ts4.8/*"] }
```
Comment (rbuckton) [inline types/node/package.json:6]:
```suggestion
        ">=4.9.0-0": { "*": ["*"] },
        "<4.9.0-0": { "*": ["ts4.8/*"] }
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/node/package.json (hunk 1 above)
Comment (sandersn) [reply types/node/package.json:6]: Actually, with `"<4.9.0-0"`, `">=4.9.0-0"` is no longer needed. I'll change dtslint to allow `<=4.8` or `<4.9.0-0`

**Comment by sandersn** [comment]
I'm going to update the format of typesVersions in dependents in a separate PR. Those are the only failures, so I'm going to merge this.

**Comment by peterblazejewicz** [comment]
would that impact CI validation time? Let me check

**Comment by pauhull** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62389 fails because of this

**Comment by peterblazejewicz** [comment]
@pauhull those errors with TS versioning have been fixed in master. You have to update your PR from current master

**Pull request merged**